internal/kitchensinktest/credential_refresh_retry_test.go

```diff
@@ -0,0 +1,60 @@
+package kitchensinktest
+
+import (
+	"context"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/internal/sdk"
+)
+
+func TestCredentialRefreshRetry(t *testing.T) {
+	sdk.NowTime = func() time.Time { return time.Unix(0, 0) }
+	defer func() {
+		sdk.NowTime = time.Now
+	}()
+
+	credvalues := []aws.Credentials{
+		{AccessKeyID: "foo", SecretAccessKey: "bar"},
+		{AccessKeyID: "baz", SecretAccessKey: "qux"},
+	}
+	mhttp := &mockHTTP{
+		resps: []*http.Response{
+			{StatusCode: 500, Body: http.NoBody},
```

Comment:
it's interesting to me that we retry by default on 500. I guess this is because of high availability in services?